pkg/controller/namespace/namespace_controller.go

```diff
@@ -0,0 +1,192 @@
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	namespaceList := &corev1.NamespaceList{}
+	opts := client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, namespaceList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get namespaceList")
+		return reconcile.Result{}, err
+	}
+
+	subjectPermissionList := &managedv1alpha1.SubjectPermissionList{}
+	opts = client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, subjectPermissionList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get clusterRoleBindingList")
+		return reconcile.Result{}, err
+	}
+
+	clusterRoleList := &v1.ClusterRoleList{}
+	opts = client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, clusterRoleList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get clusterRoleList")
+		return reconcile.Result{}, err
+	}
+
+	roleBindingList := &v1.RoleBindingList{}
+	opts = client.ListOptions{Namespace: request.Namespace}
+	err = r.client.List(context.TODO(), &opts, roleBindingList)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get rolebindingList")
+		return reconcile.Result{}, err
+	}
+
+	// loop through all subject permissions
+	// get namespaces allowed in each permission
+	// if our namespace instance is in the safeList, create rolebinding and update condition
+	for _, subjectPermission := range subjectPermissionList.Items {
+		// loop through all permissions in each
+		for _, permission := range subjectPermission.Spec.Permissions {
+			// list of all namespaces in safelist
+			safeList := controllerutil.GenerateSafeList(permission.NamespacesAllowedRegex, permission.NamespacesDeniedRegex, namespaceList)
+
+			// if namespace is in safeList, create RoleBinding
+			if namespaceInSlice(instance.Name, safeList) {
+
+				roleBinding := controllerutil.NewRoleBindingForClusterRole(permission.ClusterRoleName, subjectPermission.Spec.SubjectName, subjectPermission.Spec.SubjectKind, instance.Name)
+
+				// if rolebinding is already created in the namespace, there's nothing to do
+				if rolebindingInNamespace(roleBinding, roleBindingList) {
+					return reconcile.Result{}, nil
+				}
+
+				err := r.client.Create(context.TODO(), roleBinding)
+				if err != nil {
+					// update the condition
+					permissionUpdatedCondition := controllerutil.UpdateCondition(&subjectPermission, "Unable to create RoleBinding: "+err.Error(), permission.ClusterRoleName, true, managedv1alpha1.SubjectPermissionFailed)
+					err = r.client.Status().Update(context.TODO(), permissionUpdatedCondition)
+					if err != nil {
+						reqLogger.Error(err, "Failed to update condition.")
+						return reconcile.Result{}, err
+					}
+					reqLogger.Error(err, "Failed to create clusterRoleBinding")
+					reqLogger.Info("Failed to create rolebinding"+roleBinding.Name, "Error", err)
```

Comment:
Is there a reason we don't use `fmt.Sprintf()` here? Also why are we logging both an Error and Info level message?